detekt-report-sarif/src/test/kotlin/io/github/detekt/report/sarif/SarifOutputReportSpec.kt

```
@@ -19,9 +24,10 @@ class SarifOutputReportSpec : Spek({
 
         it("renders multiple issues") {
             val result = TestDetektion(
-                createFinding(ruleName = "TestSmellA"),
-                createFinding(ruleName = "TestSmellB"),
-                createFinding(ruleName = "TestSmellC")
+                createFinding(ruleName = "TestSmellA", severity = SeverityLevel.ERROR),
+                createFinding(ruleName = "TestSmellB", severity = SeverityLevel.WARNING),
+                createFinding(ruleName = "TestSmellC", severity = SeverityLevel.INFO),
+                createFinding(ruleName = "TestSmellD", severity = SeverityLevel.IGNORE)
```

🤔 does this have sense? Should we ever have an issue with ignore severity? I mean, to our users we provide the option to set severity as ignore. But, inside the core should we have that severity? Shouldn't we map that severity as `active = false` and don't even have it as a valid value in `SeverityLevel`?

---

Good idea 💡

---

It is a good question.  When breaking change is allowed, I meant to see if we should all migrate `active = false` to `severity = ignore`.

We can postpone that discussion later and I will remove SeverityLevel.IGNORE in this PR to not incur confusion.

---

> We can postpone that discussion later and I will remove SeverityLevel.IGNORE in this PR to not incur confusion.

+1

---

Do we need to pospone it? I mean, right now we can make that `Rule.active` is the same as `yml.active && yml.severity != "ignore"`. And we can remove `SeverityLevel.IGNORE` as a value of the enum.

---

I think about determining `severity == "ignore"` when calculating active, but the more I think, the more edge cases we could introduce. There can be a combination of cases:
- `active` is false, and `severity` is not set either => We should not run the rule
- `active` is false, but `severity` is set => We should not run the rule, but should we raise an error?
- `active` is true, and `severity` is not set => We should run the rule, with default level = warning
- `active` is true, and `severity` is set => We should run the rule, unless the default level is ignore?

Now in the future for Detekt 2.0, I would prefer using one config key to configure the severity and enabled state. Having two causes more confusion in my opinion.

---

I agree, we should remove `active` from the config file in 2.0.